### ISSUE TITLE
Wrong behavior when :dir set

### ISSUE THREAD

**Issue by akirakyle** [nnicandro/emacs-jupyter#302]
When setting the `:dir` header argument, it looks like the path is being expanded twice.

For example the following block run in a file in `/home/notebook`
```
#+BEGIN_SRC jupyter-python :dir src
print("hi")
#+END_SRC
```
will result in the error 
```
FileNotFoundErrorTraceback (most recent call last)
<ipython-input-1-1028f0c44f21> in <module>
      1 import os
      2 __JUPY_saved_dir = os.getcwd()
----> 3 os.chdir("/home/notebook/src/src")
      4 try:
      5     get_ipython().run_cell("""%load_ext autoreload

FileNotFoundError: [Errno 2] No such file or directory: '/home/notebook/src/src'
```
It appears this is due to org changing `default-directory` before `org-babel-expand-body:jupyter` is called so then `(expand-file-name dir)` is expanding again. As a patch I just replaced that call with `default-directory` and it works.

**Comment by akirakyle** [comment]
I also found another issue with setting `:dir` in `jupyter-python`. It seems `org-babel-jupyter-transform-code` isn't escaping the code correctly. Hence the following issue

```
#+BEGIN_SRC jupyter-python :session py :dir src
print(r"\r")
#+END_SRC

#+RESULTS:
:RESULTS:
# [goto error]
:   File "<ipython-input-16-37eae21e09aa>", line 1
:     print(r"
:             ^
: SyntaxError: EOL while scanning string literal
:END:
```

**Comment by nnicandro** [comment]
I can confirm both of your issues.

For the first one, I thinks it's just a matter of using `default-directory` whenever Org sets it, i.e. when the source block is being executed. I don't think we should change `org-babel-expand-body:jupyter` since it has the potential to be used outside of source block execution where I think it makes sense to handle `:dir` in the way it does.

For the second one, the fix seems to be to mark the string passed to `get_ipython().run_cell` as a raw string in the code below. That way backslashes are treated as any other character in the expanded code.

https://github.com/nnicandro/emacs-jupyter/blob/0dec83f7e4a125c73430a45c2fb2119dcc31bd28/jupyter-python.el#L98-L105

**Comment by nnicandro** [comment]
Both issues should be fixed by 113acb239d9e300a459be46be313f1fc4a264023 and a2cbebdfda1ed924eaeed673ab26a1daa91d8493.
